artifactory/commands/transferfiles/delayedartifactshandler.go

```diff
@@ -246,8 +255,10 @@ type delayUploadHelper struct {
 }
 
 // Decide whether to delay the deployment of a file by running over the shouldDelayUpload array.
-// When there are multiple levels of requirements in the deployment order, the first comparison function in the array can be removed each time in order to no longer delay by that rule.
 func (delayHelper delayUploadHelper) delayUploadIfNecessary(phase phaseBase, file api.FileRepresentation) (delayed, stopped bool) {
+	if phase.phaseId != api.Phase1 {
```

Comment:
Shouldn't files be delayed in other phases? For example:
Phase 2 - Found new docker image that contains a `manifest.json`.
Phase 3 - Handling failures, `manifest.json` can be one of them.

Reply:
You are right. Thanks